tests/conftest.py

```diff
@@ -72,6 +72,9 @@ def clean_event_loop():
                 pass
         loop.call_soon_threadsafe(loop.stop)
         RE._th.join()
+        pending = asyncio.all_tasks(loop)
+        if pending:
+            loop.run_until_complete(asyncio.gather(*pending, return_exceptions=True))
```

Comment:
Do the returned exceptions actually get propagated and raised in `run_until_complete`? The tests using this fixture will need to fail if any of those tasks do.

I think either way it's the same as this PR unfortunately:

https://github.com/bluesky/ophyd-async/issues/260#issuecomment-2100934292

We should be finding out exactly what task isn't being finished before the run engine is stopped.

Reply:
OK, we talked about this over slack and decided to solve this in #519 instead